### PR TITLE
:tada: Source Amplitude:  added missing attrs in events schema, enabled default availability strategy

### LINKED ISSUE #22248: Source source-amplitude: implement AvailabilityStrategy
## What
We want GA connectors to make use of an AvailabilityStrategy in their streams to be able to perform error handling on the
stream level. The base use case supported by the `HttpAvailabilityStrategy` is for stream-based permissions,
in order to be able to skip a stream that the connector no longer has access to (because of e.g. scope
changes on an Oauth user config). An `AvailabilityStrategy` can be used to address any error that may occur
on a stream level, to avoid hitting errors, or having to except the same error for every record in a stream,
instead of skipping that stream altogether.

## How
There are a few different ways to go about this, usually depending on how much custom error handling we already
do in a connector, but also due to how the connector works.
- If we have no custom error handling: Inherit the default HttpAvailabilityStrategy. This is as simple as removing the override setting that currently sets
  the AvailabilityStrategy for the streams to `None`. This is useful if we don't currently do much custom error handling
  in the connector.
- If we have custom error handling for 403: Forbidden errors: Do the above to inherit the HttpAvailabilityStrategy,
  removing the custom error handling, but continuing any testing we have that 403s are handled gracefully by the connector
- If we have custom error handling for lots of per-stream errors: Extend the HttpAvailabilityStrategy. You can subclass
  the HttpAvailabilityStrategy to change the `reasons_for_unavailable_status_codes` method to except more error codes and
  provide messages for those error codes.
- If there is a smarter way to determine availability than trying to read the first record and excepting errors: Inherit
  the AvailabilityStrategy base class. For example, if there are clearly defined scopes required to access each stream in
  a given source, you might instead determine whether a stream is available by checking the scopes on the authenticator to
  see if it has the scopes required by the stream.

Note that the AvailabilityStrategy is defined on each stream. For most cases, it is easiest to set it in your stream's
base stream class, where the override is, but for some use cases, different streams might have different availability strategies -
and that's totally fine.

### LINKED ISSUE #24480: Source Amplitude: Add undeclared columns to spec
## What
In this [PR](https://github.com/airbytehq/airbyte/pull/23985), `BasicReadTest.test_read` test in the Connector Acceptance Tests was updated to fail if the connector produces
stream records which contain columns that haven't been declared in the spec.

Amplitude currently fails the updated test. Its `acceptance-test-config.yml` was edited with the
`fail_on_extra_columns: false` parameter in order to avoid this change from making the connector fail CAT.

We want to add the undeclared columns to the spec.

**How this will help**:
- Column selection is currently blocked for connectors that don't declare all columns that they pass. This
  is because turning on column selection will stop these undeclared columns from being sent, when they were
  previously being sent. Doing this allows us to enable column selection for Amplitude!
- Users will better understand the shape of the data they'll receive, since the columns will match the spec.

## How
The following descriptions of streams that pass undeclared columns come from results of the failed connector acceptance test:

```
columns.txt:./source-amplitude.txt-85-The events stream has the following schema errors:
columns.txt:./source-amplitude.txt:86:Additional properties are not allowed ('$insert_key', 'data_type', 'plan', 'source_id', 'partner_id', 'global_user_properties' were unexpected)

```

1. Add the missing properties indicated by the `Additional properties are not allowed ('<column>', 'column' were unexpected)` logs to the connector's spec.
2. Remove `fail_on_extra_columns: false` from the `acceptance-test-config.yml` file.
3. Commit changes to spec and `acceptance-test-config.yml` and open a PR.
4. Run tests on the connector, either automatically via CI or manually via the `/test` command
5. Profit!

Definition of done: Amplitude passes CAT without declaring `fail_on_extra_columns: false`. If the
API starts sending over extra columns in the future, we will catch and fix them as part of the #connector-health
movement.

### LINKED ISSUE #24941: Amplitude Improvement Epic

### PR DESCRIPTION
**Pull request by midavadim**
added missing attrs in events schema, 
enabled default avilability strategy

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Review by artem1205 — CHANGES_REQUESTED** [review]
availability strategy was removed, because for this stream API may return 404 (see [docs](https://www.docs.developers.amplitude.com/analytics/apis/export-api/#response) ) during read of first slice.

**Comment by midavadim** [comment]
> availability strategy was removed, because for this stream API may return 404 (see [docs](https://www.docs.developers.amplitude.com/analytics/apis/export-api/#response) ) during read of first slice.

@artem1205 
404 error is caught at line 163 and stream just returns empty list of records.
So this error does not get to 'availability strategy' handler

**Comment by artem1205** [comment]
Yep, but this means that availability strategy is useless here, isn't it?

**Comment by midavadim** [comment]
> Yep, but this means that availability strategy is useless here, isn't it?

I suppose it can handle another types of errors, what do you think?

**Review by artem1205 — APPROVED** [review]

File: airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml [hunk 1]
```diff
@@ -30,7 +30,6 @@ acceptance_tests:
           extra_fields: no
           exact_order: no
           extra_records: yes
-        fail_on_extra_columns: false
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-amplitude/acceptance-test-config.yml:33]: Woot 🥳

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/streams.py [hunk 2]
```diff
@@ -68,10 +67,6 @@ def url_base(self) -> str:
         subdomain = "analytics.eu." if self.data_region == "EU Residency Server" else ""
         return f"https://{subdomain}amplitude.com/api/"
 
-    @property
-    def availability_strategy(self) -> Optional["AvailabilityStrategy"]:
-        return None
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/streams.py:73]: Just double checked - I think @midavadim is right. The availability strategy's `get_first_record_for_slice` calls `stream.read_records` so the 404 error handling that returns empty records should work, and the AS should say "connected to stream, but got 0 records". So this should work in that case 👍🏻

**Review by erohmensing — COMMENTED** [review]
Nice work! Could you please also add the missing column declarations to the cohorts stream? Here are all of the missing columns from the past week (they look pretty much teh same last month) 

image

This comes from the "please check the datadog dashboard" note [here](https://github.com/airbytehq/airbyte/issues/24480#issuecomment-1488885878).  I guess our test doesn't catch it because `cohorts` is declared as an empty stream.

**Review by maxi297 — APPROVED** [review]
LGTM given Ella's concern is addressed!

**Comment by erohmensing** [comment]
@midavadim that's fine with me! If any of the types are wrong, that's on the documentation, not us. And we'll catch them in the schema validation errors. Thank you! ⭐

**Review by erohmensing — APPROVED** [review]

**Comment by erohmensing** [comment]
Then I think we can safely close #24480 as well :)

**Pull request merged**